src/components/OnboardingDialog/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react'
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import { radius, shadow, space, duration, transition } from '../../theme'
+import { useTransition } from '../../hooks'
+
+const BodyWrapper = styled.div`
+  background-color: white;
+  border-radius: ${radius.lg};
+  box-shadow: ${shadow.strong};
+  padding: ${space[32]};
+`
+
+const OnboardingDialogBody = ({ children }) => (
+  <BodyWrapper>{children}</BodyWrapper>
+)
+
+OnboardingDialogBody.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+const DialogWrapper = styled.div`
+  position: absolute;
+  top: ${props => space[props.positionTop]};
+  left: 0;
+  right: 0;
+  z-index: 1;
+`
+
+const OnboardingDialog = ({ children, show, positionTop }) => {
+  if (!show) return null
+
+  return <DialogWrapper positionTop={positionTop}>{children}</DialogWrapper>
+}
+
+OnboardingDialog.defaultProps = {
+  positionTop: 56,
+}
+
+OnboardingDialog.propTypes = {
+  positionTop: PropTypes.number,
+  show: PropTypes.bool.isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+const Container = styled.div`
+  position: relative;
+`
+
+const Backdrop = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgba(255, 255, 255, 0.8);
+  opacity: ${props =>
+    props.state === 'entering' || props.state === 'entered' ? 1 : 0};
+  transition: opacity ${transition};
+`
+
+const OnboardingWrapper = ({ children, active, state }) => {
+  return (
+    <Container>
+      {children}
+      {active && <Backdrop state={state} />}
+    </Container>
+  )
+}
+
+OnboardingWrapper.propTypes = {
+  active: PropTypes.bool.isRequired,
+  state: PropTypes.string.isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+const Onboarding = ({ children, defaultOn }) => {
+  const onboardingProps = useOnboarding({
+    defaultOn,
+  })
+
+  return children(onboardingProps)
+}
+
+Onboarding.defaultProps = {
+  defaultOn: true,
+}
+
+Onboarding.propTypes = {
+  defaultOn: PropTypes.bool.isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+function useOnboarding(config = {}) {
+  const { defaultOn = true, defaultTransitionState = 'exited' } = config
```

Comment:
`exited` should be a const, see also other implementations. Maybe we can import them from `useTransition`

Reply:
Btw we then could use those consts in the propType validation as stated above as well.